## Code Review: PR #10274 (Round 1)

### Summary

Small, surgical fix for #10272. The new `disconnectConnector$` command wraps `deleteConnector$` and clears `internalJustConnectedTypes$` after a successful DELETE so the Connectors page doesn't keep a stale card in the Connected section. Fix is minimal, no try/catch, identity-stable `Set` update, and no AP-4 mock violations. Tests go through `bootstrap$` via `detachedSetupPage` and assert observable behavior (set membership + DOM).

### Key Findings

#### Critical Issues (P0)

- **`turbo/apps/platform/src/signals/zero-page/settings/__tests__/disconnect-clears-just-connected.test.ts:36` and `:40` — blocking `lint-eslint` on CI.** `oxlint`'s `eslint-plugin-vitest` rules `prefer-to-be-truthy` / `prefer-to-be-falsy` require `.toBeTruthy()` / `.toBeFalsy()` instead of `.toBe(true)` / `.toBe(false)`. CI log (`lint-eslint` job `72100458190`):
  ```
  ##[error]Use `toBeTruthy` instead.
  ##[error]Use `toBeFalsy` instead.
  Found 0 warnings and 2 errors.
  @vm0/app#lint: command exited (1)
  ```

#### High Priority (P1)

- None.

### Testing Review

#### Coverage

- Signal-level regression: `__tests__/disconnect-clears-just-connected.test.ts` pins the core invariant that `justConnectedTypes$` no longer contains the type after `disconnectConnector$`. PR description states it was verified FAIL-on-main / PASS-on-branch.
- UI integration: `zero-connectors-page-disconnect.test.tsx` drives the full api-token connect → disconnect flow and asserts the card leaves the Connected section.
- Gap: no OAuth-path end-to-end test (`connectConnector$` → `disconnectConnector$`). Acceptable because both producers funnel through the same `internalJustConnectedTypes$` mutation — mentioning only as P2.

#### Convention Compliance

- Mocks at boundary (MSW via `mockConnectors`), not `vi.mock` with relative paths. Clean.
- `detachedSetupPage` invokes `bootstrap$` like production — no Bad Smell #19 violation.
- No `any`, no try/catch, no fake timers, no dynamic imports, no hardcoded URLs.
- Lint violations listed under P0 above.

#### Testing Verdict: Convention Violations (the P0 lint errors)

### Verdict: Changes Requested

Fixing P0 and will re-review.

---
*Round 1 of automated review-fix loop*

--- a/turbo/apps/platform/src/signals/zero-page/settings/__tests__/disconnect-clears-just-connected.test.ts
+++ b/turbo/apps/platform/src/signals/zero-page/settings/__tests__/disconnect-clears-just-connected.test.ts
@@ -33,10 +33,10 @@ describe("deleteConnector$ + justConnectedTypes$", () => {
       context.signal,
     );
 
-    expect(context.store.get(justConnectedTypes$).has("ahrefs")).toBe(true);
+    expect(context.store.get(justConnectedTypes$).has("ahrefs")).toBeTruthy();
 
     await context.store.set(disconnectConnector$, "ahrefs", context.signal);
 
-    expect(context.store.get(justConnectedTypes$).has("ahrefs")).toBe(false);
+    expect(context.store.get(justConnectedTypes$).has("ahrefs")).toBeFalsy();
   });
 });

